core/jobs/batch_jobs/exp_version_history_computation_job.py

```diff
@@ -0,0 +1,1230 @@
+                        break
+                verified_state_vh.append(expected_state_vh)
+                verified_metadata_vh.append(expected_metadata_vh)
+
+        return (exp_id, verified)
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        all_explorations_vlatest = (
+            self.pipeline
+            | 'Get all ExplorationModels' >> ndb_io.GetModels(
+                exp_models.ExplorationModel.get_all(include_deleted=False)
+            )
+            | 'Create key-value pairs with id and exp models' >>
+                beam.Map(lambda model: (model.id, model))
+        )
+
+        all_explorations_v1 = (
+            all_explorations_vlatest
+            | 'Get all the snapshot models' >> ndb_io.GetModels(
+                exp_models.ExplorationSnapshotContentModel.get_all(
+                    include_deleted=False
+                )
+            )
+            | 'Filter the snapshot models without None' >>
+                beam.Filter(lambda model: model is not None)
+            | 'Get the snapshot models at version 1' >> beam.Filter(
+                lambda model: model.get_version_string() == '1'
+            )
+            | 'Get reconstituted explorations at v1' >>
+                beam.Map(self.generate_exploration_from_snapshot)
+            | 'Create key-value pairs with id and exp models at v1' >>
+                beam.Map(lambda model: (model.id, model))
+        )
+
+        all_commit_logs = (
+            all_explorations_v1
+            | 'Get all ExplorationCommitLogEntryModels' >> ndb_io.GetModels(
+                exp_models.ExplorationCommitLogEntryModel.get_all(
+                    include_deleted=False
+                )
+            )
+            | 'Create key-value pairs with id and commit log models' >>
+                beam.Map(lambda model: (model.exploration_id, model))
+        )
+
+        all_version_history_models = (
+            all_commit_logs
+            | 'Get all ExplorationVersionHistoryModels' >>
+                ndb_io.GetModels(
+                    exp_models.ExplorationVersionHistoryModel.get_all(
+                        include_deleted=False
+                    )
+                )
+            | 'Create key-value pairs with id and version history models' >>
+                beam.Map(lambda model: (model.exploration_id, model))
+        )
+
+        verification_results = (
+            ({
+                'exp_models_v1': all_explorations_v1,
+                'exp_models_vlatest': all_explorations_vlatest,
+                'commit_log_models': all_commit_logs,
+                'version_history_models': all_version_history_models
+            })
+            | 'Group by key' >> beam.CoGroupByKey()
+            | 'Get rid of exploration id' >>
+                beam.Values() # pylint: disable=no-value-for-parameter
+            | 'Filter valid model groups' >> beam.Filter(
+                self.filter_valid_model_group
```

Comment:
Should we also count then number of invalid groups?

Reply:
Done.